stan/math/opencl/multiply.hpp

```diff
@@ -75,15 +76,16 @@ inline matrix_cl<return_type_t<T1, T2>> multiply(const matrix_cl<T1>& A,
           / wgs);
   try {
     if (split <= 1) {
-      opencl_kernels::matrix_multiply(
-          cl::NDRange(Mpad, Npad / wpt), cl::NDRange(local, local / wpt), A, B,
-          temp, A.rows(), B.cols(), B.rows(), A.view(), B.view());
+      opencl_kernels::matrix_multiply(cl::NDRange(Mpad, Npad / wpt),
+                                      cl::NDRange(local, local / wpt), A.eval(),
+                                      B.eval(), temp, A.rows(), B.cols(),
+                                      B.rows(), A.view(), B.view());
     } else {
       matrix_cl<return_type_t<T1, T2>> tempSplit(A.rows(), B.cols() * split);
       opencl_kernels::matrix_multiply(cl::NDRange(Mpad, Npad / wpt, split),
-                                      cl::NDRange(local, local / wpt, 1), A, B,
-                                      tempSplit, A.rows(), B.cols(), B.rows(),
-                                      A.view(), B.view());
+                                      cl::NDRange(local, local / wpt, 1),
+                                      A.eval(), B.eval(), tempSplit, A.rows(),
+                                      B.cols(), B.rows(), A.view(), B.view());
```

Comment:
Do these evals need to be called before every custom kernel?

Reply:
Kernels need buffer objects, which are in `matrix_cl`. This eval is cheap for `arena_matrix_cl`. If you want I can instead do these evals at start of the function into a new variable (there is always exactly one kernel being run that needs evaled arguments).

Reply:
Nah it's fine as is